### PR TITLE
Generate list of excluded samples

### PR DESCRIPTION
**Pull request by ellisrichardj**
This PR adds the functionality to generate the list of samples that are excluded from the phylogenetic analysis at each stage of the filtering.  The new table simply contains the submission number and the reason the sample has been excluded: 
- Low quality data
- Impure culture
- Not M.bovis
- Identified Outlier

The tests have also been updated to cover the new function

### CODE REVIEW

File: bin/listExcluded.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    outliers_df = pd.read_csv(outliers_list, dtype='object', header=None)
+    outliers_df.columns = ['Submission']
+    # add column to indicate that sample has been identified as an outlier
+    outliers_df['Exclusion'] = 'Identified Outlier'
+
+    allClean_df = pd.read_csv(allclean_csv, dtype='object')
+    allClean_df.columns.values[0] = "Submission"
+    # add column with reason for exclusion
+    allClean_df['Exclusion'] = allClean_df.apply(lambda x: exclusionBasis(x['ID'], x['Outcome']), axis=1)
```
Comment (nick-pestell) [inline bin/listExcluded.py:33]: I like this!
😎

**Review by nick-pestell — CHANGES_REQUESTED** [review]
This looks great! Thanks Richard.

A few tiny comments.

One thing, can I check, is their a hierarchy of exclusion reason, i.e. if there is a sample which is excluded for multiple reasons, i.e. it has high-N but is also impure culture, what would be the reason that is reported.
If I understand the code correctly, you would have, "Not M.bovis", then "Impure culture", then "Low quality data" and finally, "Identified outlier". I think this is probably by design and is what we would want?

**Comment by ellisrichardj** [comment]
> This looks great! Thanks Richard.
> 
> A few tiny comments.
> 
> One thing, can I check, is their a hierarchy of exclusion reason, i.e. if there is a sample which is excluded for multiple reasons, i.e. it has high-N but is also impure culture, what would be the reason that is reported. If I understand the code correctly, you would have, "Not M.bovis", then "Impure culture", then "Low quality data" and finally, "Identified outlier". I think this is probably by design and is what we would want?

Yes, the exclusion reasons are hierarchical by design and this reflects the order in which they are excluded during the processing

File: bin/listExcluded.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import pandas as pd
+import argparse
+from datetime import date
+
+
+# determine if sample is not Mbovis or is contaminated
+def exclusionBasis(ID, Outcome):
+    if ID != 'Mycobacterium bovis':
+        return 'Not M.bovis'
```
[the remark this answers is not in the document]
Comment (ellisrichardj) [reply bin/listExcluded.py]: Sure - no problem

**Review by nick-pestell — APPROVED** [review]
Perfect! Thanks for making those changes! :)

**Pull request merged**